app/models/coupon.rb

```diff
@@ -1,4 +1,4 @@
 class Coupon < ApplicationRecord
   validates :amount, :name, presence: true
-  has_many :order_items, as: :source
+  has_many :order_items, as: :source, dependent: :destroy
```

Comment:
valid but you should make sure that you don't destroy order items for completed orders. If someone has been charged for something you have to keep a record of it forever.